pkg/operator/controller/externaldns/deployment.go

```diff
@@ -201,8 +201,7 @@ func desiredExternalDNSDeployment(cfg *deploymentConfig) (*appsv1.Deployment, er
 	}
 
 	nodeSelectorLbl := map[string]string{
-		osLabel:             linuxOS,
-		masterNodeRoleLabel: "",
```

Comment:
I think I added this label maybe because I saw it on some cluster operators' operands. I don't remember any strong reason for it and we cannot come up with any. But we see problems related to it. RIP then.